ably/ably_test.go

```diff
@@ -112,6 +113,85 @@ func assertNil(t *testing.T, object interface{}) {
 	}
 }
 
+// ObjectsAreEqual determines if two objects are considered equal.
+//
+// This function does no assertion of any kind.
+func objectsAreEqual(expected, actual interface{}) bool {
```

Comment:
This function can be replaced by just `reflect.DeepEqual`. See https://play.golang.org/p/rQOdjzcqpaD